src/Protocol/Protocol_1_8.cpp

```diff
@@ -2554,28 +2554,88 @@ void cProtocol_1_8_0::HandlePacketPlayerPosLook(cByteBuffer & a_ByteBuffer)
 
 void cProtocol_1_8_0::HandlePacketPluginMessage(cByteBuffer & a_ByteBuffer)
 {
+	// https://wiki.vg/index.php?title=Plugin_channels&oldid=14089#MC.7CAdvCmd
+
 	HANDLE_READ(a_ByteBuffer, ReadVarUTF8String, AString, Channel);
 
+	const std::string_view ChannelView = Channel;
+
 	// If the plugin channel is recognized vanilla, handle it directly:
-	if (Channel.substr(0, 3) == "MC|")
+	if (ChannelView.substr(0, 3) == "MC|")
 	{
-		HandleVanillaPluginMessage(a_ByteBuffer, Channel);
```

Comment:
The control flow is clearer if this is a seperate function, rather than having a single function be 80 lines long. It certainly doesn't need to be exposed though, it would be fine to make it a private/static function.

Reply:
Yep, I added it back and made the virtual function useful, all the 1.8 - 1.12 messages should be read now.